common/persistence/elasticsearch/esVisibilityStore.go

```diff
@@ -570,12 +571,87 @@ func addMustQuery(dsl *fastjson.Value, queryString string) {
 	valOfTopQuery.Set("bool", fastjson.MustParse(newValOfBool))
 }
 
+func (v *esVisibilityStore) processSortField(dsl *fastjson.Value) (string, error) {
+	isSorted := dsl.Exists(dslFieldSort)
+	var sortField string
+
+	if !isSorted { // set default sorting by StartTime desc
+		dsl.Set(dslFieldSort, fastjson.MustParse(jsonSortForOpen))
+		sortField = definition.StartTime
+	} else { // user provide sorting using order by
+		// sort validation on length
+		if len(dsl.GetArray(dslFieldSort)) > 1 {
+			return "", errors.New("only one field can be used to sort")
+		}
+		// sort validation to exclude IndexedValueTypeString
+		obj, _ := dsl.GetArray(dslFieldSort)[0].Object()
+		obj.Visit(func(k []byte, v *fastjson.Value) {
```

Comment:
How about adding a comment here which says fastjson doesn't support get keys of an object? People may be confused why there's only one key in this object and Visit() is used.

Reply:
added